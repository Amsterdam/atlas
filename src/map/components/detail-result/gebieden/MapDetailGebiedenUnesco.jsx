import React from 'react';
import PropTypes from 'prop-types';

const MapDetailGebiedenUnesco = ({ panoUrl, unesco }) => (
  <section className="map-detail-result">
    <header
      className={`
        map-detail-result__header
        map-detail-result__header--${panoUrl ? 'pano' : 'no-pano'}
      `}
    >
      {panoUrl && (
        <img
          alt="Panoramabeeld"
          className="map-detail-result__header-pano"
          src={panoUrl}
        />
      )}
      <div className="map-detail-result__header-container">
        <h1 className="map-detail-result__header-title">Unesco</h1>
        <h2 className="map-detail-result__header-subtitle">{unesco.label}</h2>
      </div>
    </header>
  </section>
);

MapDetailGebiedenUnesco.defaultProps = {
  panoUrl: ''
};

MapDetailGebiedenUnesco.propTypes = {
  unesco: PropTypes.shape({
    label: PropTypes.string.isRequired
  }).isRequired,
  panoUrl: PropTypes.string
};

export default MapDetailGebiedenUnesco;
