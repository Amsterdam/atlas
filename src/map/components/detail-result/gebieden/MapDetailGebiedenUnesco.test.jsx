import React from 'react';
import { shallow } from 'enzyme';

import MapDetailGebiedenUnesco from './MapDetailGebiedenUnesco';

describe('MapDetailGebiedenUnesco', () => {
  it('should render everything', () => {
    const unesco = {
      label: 'Unesco label'
    };
    const wrapper = shallow(
      <MapDetailGebiedenUnesco
        panoUrl="panoUrl"
        unesco={unesco}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
