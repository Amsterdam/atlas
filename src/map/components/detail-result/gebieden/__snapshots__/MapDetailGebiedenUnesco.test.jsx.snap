// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailGebiedenUnesco should render everything 1`] = `
<Connect(MapDetailResultWrapper)
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Unesco label"
  title="UNESCO"
/>
`;
